### ISSUE TITLE
Editing date with widgets generates "invalid date"

### ISSUE THREAD

**Issue by unode** [Extended-Thunder/send-later#630]
Steps to reproduce:
1. In an email draft click Send Later
2. Select a day with the day widget selector - pick tomorrow
3. Select an hour by typing the number manually - type `8` <kbd>TAB</kbd> `0` <kbd>TAB</kbd> `A` (for AM)
4. Click outside the Send Later floating window
5. Click Send Later again

After 5. the date shows as `mm/dd/yyyy`, the hour `--:-- --` and the "Send at" button shows `Enter a valid date above`.

6. Removing `AM` from the `Send at:` textual field makes the widget recognize the date and the Send button shows the correct time estimate.

I don't have any `LC_` variables defined.
Thunderbird is version 102.14.0
Send Later is [v10.1.5](https://github.com/Extended-Thunder/send-later/releases/tag/v10.1.5)

# Screenshots

###  After manually setting `AM` in the widget

![screenshot_2023-09-05_00-53-25_697618764](https://github.com/Extended-Thunder/send-later/assets/122319/c0babc37-f23f-42b5-b671-f18acc688546)

### After closing and opening the Send Later dialog when a time and date had been previously selected using the widgets

![screenshot_2023-09-05_00-57-23_873687657](https://github.com/Extended-Thunder/send-later/assets/122319/764e3654-b4e3-4f68-bb72-a8b7d6e0ce98)

### After deleting `AM` in the `Send at` field.

![screenshot_2023-09-05_00-53-44_929773814](https://github.com/Extended-Thunder/send-later/assets/122319/5b6fd57c-cb14-4db3-a145-34dbf14c09a0)

**Comment by jikamens** [comment]
This is not the behavior I see. See the attached video below.

>I don't have any LC_ variables defined.

Regardless, this is a locale issue of some sort, because whatever locale Send Later is parsing dates in doesn't understand the AM.

If you install version 10.1.6 of Send Later (instructions below) and then restart and search for "locale" in your error console (type Ctrl-Shift-J to open it and then enter locale in the filter box at the top), you should see a line from Send Later showing what your  configured UI and OS locales are. My guess is that your OS locale is one where the date parser doesn't understand how to parse "AM".

In any case you can work around this problem by enabling customized date formats in the Send Later preferences and setting the short format to this: `%m/%d/%Y, %R` . This will force the time that gets inserted into the text box to be in 24-hour format. Though if you really want to eliminate ambiguity you might want to use `%F`, instead of `%m/%d/%Y`, since `%F` will give you YYYY-MM-DD format.

I'm closing this issue because to my knowledge the add-on is behaving as designed and in any case the date format customization is exactly intended to provide a workaround for issues like this, so a fix is available already.
 
[Screencast from 2023-09-04 21-21-36.webm](https://github.com/Extended-Thunder/send-later/assets/1598067/056ec447-67da-40ab-8bb4-bc94ba992303)

To install 10.1.6:

* Download [this file](https://github.com/Extended-Thunder/send-later/releases/download/v10.1.6/send_later.xpi) to your computer. (If you are using Firefox, make sure to right click on the link and select
"Save Link As..." rather than just clicking on it, because otherwise Firefox will try to install it as a Firefox add-on.)
* Open the Add-ons page in Thunderbird ("Add-ons and Themes" from the corner "hamburger" menu).
* Click the gear icon at the top and select "Install Add-on From File...".
* Browse to and select the downloaded file.
* Click through the installation pop-ups.
* You can delete the downloaded file once you've installed it into Thunderbird.

**Comment by unode** [comment]
Shown now but:

```
Send Later version 10.1.6 on Thunderbird 102.14.0 (20230903124808) [linux x86-64] UI locale en-US, Date Locale en-US [tools.js:59:13](moz-extension://50e7363c-6f0f-4e25-bf08-d2d78d94251d/utils/tools.js)
```

**Comment by jikamens** [comment]
That's fascinating.

I have no idea why it's behaving this way.

Reopening the issue to remind me to give it some more thought and try to figure it out.

Nevertheless the workaround I suggested above should work for you. Let me know if it doesn't.

**Comment by jikamens** [comment]
This issue should be fixed in release 10.1.7 of Send Later. To check what version of Send Later you have:

* Open the Add-ons page in Thunderbird ("Add-ons and Themes" from the corner "hamburger" menu).
* Click on Extensions on the left if you're not already on the left if you're not already there.
* Find Send Later in the list of extensions and click on it to open its details page.
* The version number should be displayed there next to
"Version".

This release is not yet available for download from addons.thunderbird.net, but you can download and install it from its [GitHub release page](https://github.com/Extended-Thunder/send-later/releases/tag/v10.1.7) as follows:

* Download [this file](https://github.com/Extended-Thunder/send-later/releases/download/v10.1.7/send_later.xpi) to your computer. (If you are using Firefox, make sure to right click on the link and select
"Save Link As..." rather than just clicking on it, because otherwise Firefox will try to install it as a Firefox add-on.)
* Open the Add-ons page in Thunderbird ("Add-ons and Themes" from the corner "hamburger" menu).
* Click the gear icon at the top and select "Install Add-on From File...".
* Browse to and select the downloaded file.
* Click through the installation pop-ups.
* You can delete the downloaded file once you've installed it into Thunderbird.

Note that this is a prerelease. While we make every effort to ensure that prereleases are stable, they are a bit more likely to have bugs, so proceed with caution.

If you prefer to wait, the release will eventually be available on addons.thunderbird.net, and unless you've turned off automatic add-on updates, Thunderbird will eventually update to the new version automatically. To update immediately once it's available:

* Go to [addons.thunderbird.net](https://addons.thunderbird.net/thunderbird/addon/send-later-3/versions/) and confirm that release 10.1.7 is available for download there.
* Open the Add-ons page in Thunderbird ("Add-ons and Themes" from the corner "hamburger" menu).
* Click on Send later.
* Click the gear icon and select "Check for Updates".
* The add-on should update and then you should see the message, "Your add-ons have been updated," next to the gear icon.

Please let me know if you still see this issue in this release.

**Comment by unode** [comment]
Confirmed, bug is now gone.
